widget/widgetAssets/jsonToGrid.service.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+(function () {
+    angular
+        .module('fortisoar.marketplace')
+        .factory('jsonToGridService', jsonToGridService);
+
+    jsonToGridService.$inject = ['$http', 'API', '$q', '$window', 'toaster', '$uibModal'];
+
+    function jsonToGridService($http, API, $q, $window, toaster, $uibModal) {
+        var service = {
+            launchStandaloneWidget: launchStandaloneWidget
+        };
+
+        return service;
+
+        function launchStandaloneWidget(widgetName, widgetVersion, playbook, executeWithRecord, selectedRows) {
+          var defer = $q.defer();
+          let windowClass = {
+            'Half Width': 'mp-detail-panel-view',
+            'Full Width': 'no-animation modal-ingestion',
+            'Pop-Up': 'modal no-animaiton'
+          };
+          let param = {
+            name: widgetName,
+            version: widgetVersion,
+            installed: true
+          };
+          $http.post(API.BASE + 'widgets/detail', param).then(function (response) {
+            if (response.data['hydra:totalItems'] === 0) {
```

Comment:
add check response.data['hydra:totalItems'] to handle case where response is empty []